### PR TITLE
scope reduction construct test

### PR DESCRIPTION
**Pull request by mjcarr458**
first iteration of scope construct, based off of 5.1 example sheet. scope construct currently not implemented

### CODE REVIEW

**Comment by mjcarr458** [comment]
Updated tests with suggested comments. The test should be designed to reduce an array using scope reduction clause based off example in OpenMP example sheet.

File: tests/5.1/scope/test_scope_construct.c [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+	int s = 0;
+	int errors = 0;
+	for (int i; i < N; i++){
+		a[i] = 1;
+	}
+	OMPVV_TEST_OFFLOADING;
+	OMPVV_TEST_AND_SET_VERBOSE(errors, test_scope() != 0);
+	OMPVV_REPORT_AND_RETURN(errors);
+	test_scope(N,a,s);
+	return s;
```
Comment (tob2) [inline tests/5.1/scope/test_scope_construct.c:55]:
```suggestion
	OMPVV_TEST_AND_SET_VERBOSE(errors, test_scope(N,a,s) != 0);
	OMPVV_REPORT_AND_RETURN(errors);
```

That is: Pass the arguments to the function – and remove the now unreachable second function call and return.

File: tests/5.1/scope/test_scope_reduction_construct.c [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+	int a[N];
+	int s = 0;
+	int errors = 0;
+	for (int i; i < N; i++){
+		a[i] = 1;
+	}
+	OMPVV_TEST_OFFLOADING;
+	OMPVV_TEST_AND_SET_VERBOSE(errors, test_scope() != 0);
+	OMPVV_REPORT_AND_RETURN(errors);
+	return s;
```
Comment (tob2) [inline tests/5.1/scope/test_scope_reduction_construct.c:54]:
```suggestion
	OMPVV_TEST_AND_SET_VERBOSE(errors, test_scope(N,a,s) != 0);
	OMPVV_REPORT_AND_RETURN(errors);
```

**Review by tob2 — COMMENTED** [review]
Question: What should be the difference between those two files, except for the file name?

Both files do a reduction (hence, the now suggested 'firstprivate→shared' change) – and the only difference 'diff' shows is that one file contains a second 'test_scope' call (which is unreachable as placed after the return macro).

File: tests/5.1/scope/test_scope_reduction_construct.c (hunk 2 above)
Comment (mjcarr458) [reply tests/5.1/scope/test_scope_reduction_construct.c]: > 

Sorry about this confusion

> Question: What should be the difference between those two files, except for the file name?
> 
> Both files do a reduction (hence, the now suggested 'firstprivate→shared' change) – and the only difference 'diff' shows is that one file contains a second 'test_scope' call (which is unreachable as placed after the return macro).

Sorry about this confusion, test_scope_construct.c was left over from a previous test iteration before we decided to split scope tests. It should be removed now.

**Review by spophale — APPROVED** [review]

**Review by krishols — APPROVED** [review]

**Pull request merged**